bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/ConfigUtil.java

```diff
@@ -63,9 +66,51 @@ public static Object normalizeType(Object value, ConfigDescriptionParameter conf
         if (configDescriptionParameter != null) {
             Normalizer normalizer = NormalizerFactory.getNormalizer(configDescriptionParameter);
             return normalizer.normalize(value);
-        } else {
-            return value instanceof Double ? BigDecimal.valueOf((Double) value) : value;
+        } else if (value == null || value instanceof Boolean || value instanceof String
+                || value instanceof BigDecimal) {
+            return value;
+        } else if (value instanceof Number) {
+            return new BigDecimal(value.toString());
+        } else if (value instanceof Collection) {
+            return normalizeCollection((List) value);
         }
+        throw new IllegalArgumentException(
+                "Invalid type '{" + value.getClass().getCanonicalName() + "}' of configuration value!");
+    }
+
+    private static List normalizeCollection(List collection) {
```

Comment:
See above, I think this should be changed to
`private static List normalizeCollection(Collection collection)`